patches/PRESUBMIT.py.patch

```diff
@@ -0,0 +1,9 @@
+diff --git a/PRESUBMIT.py b/PRESUBMIT.py
+index 81b1b62c3ef84dedaba01ef067bafc66d9ebfbb4..7e4fb471f68b60b23b03603902f33fb0e90ca204 100644
+--- a/PRESUBMIT.py
++++ b/PRESUBMIT.py
+@@ -5845,3 +5845,4 @@ def CheckPythonShebang(input_api, output_api):
+                 "Please use '#!/usr/bin/env python/2/3' as the shebang of %s" %
+                 file))
+     return result
++if not globals().get('__name__'): from lib.import_inline import inline_module; inline_module('chromium_presubmit_overrides', globals(), locals())
```

Comment:
not sure I understand what's going on here, why the check for `__name__`? Seems like in general we could use this method to eliminate other python patches?

Reply:
oh, I guess `.get('__name__')` is equivalent to `__name__`

Reply:
`if not globals().get('__name__')` check is very special and it make sense only in the context of presubmit machinery.
What's exactly going on here:
1. `PRESUBMIT.py` is executed via custom python's exec/compile process, which makes `__name__` not defined in the `globals()` scope. I use this as a trigger to inject our presubmit changes.
2. `PRESUBMIT.py` can run `PRESUBMIT_test.py` if it detects that it was changed.
3. When `PRESUBMIT_test.py` is running, it `import PRESUBMIT` as usual, which makes `__name__` defined, in this case we don't inject our changes. Otherwise, some chromium tests fail, because we disable some upstream checks (expected).